ocaml/xapi/xapi_host.ml

```diff
@@ -448,7 +448,16 @@ let get_vms_which_prevent_evacuation_internal ~__context ~self ~ignore_ha =
 
 (* New Orlando style function which returns a Map *)
 let get_vms_which_prevent_evacuation ~__context ~self =
-  get_vms_which_prevent_evacuation_internal ~__context ~self ~ignore_ha:false
+  let vms =
+    get_vms_which_prevent_evacuation_internal ~__context ~self ~ignore_ha:false
+  in
+
+  let log (vm, reasons) =
+    debug "%s: VM %s preventing evacuation of host %s: %s" __FUNCTION__
+      (Ref.string_of vm) (Ref.string_of self)
```

Comment:
These will all be database references, making it hard to debug, but better than nothing.

Reply:
Yes, it would be better to log UUIDs. We currently have a mixed bag in the logs in general, but whenever I see a ref in the logs when triaging something I get disappointed. Would be good to add a `Db.VM.get_uuid` here.

Reply:
I hear ya. I thought Refs are internal and UUIDs are for the API but can fish them out for logging.